refactor: move sign message to side-panel

alfonsobries opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure you're familiar with and follow the instructions in the [contributing guidelines](https://ark.dev/docs/program-incentives/guidelines/contributing).

Please fill out the information below to expedite the review and (hopefully) merge of your pull request!
-->

## Summary

https://app.clickup.com/t/86dwuk8u2

## Checklist

<!-- Have you done all of these things?  -->

-   [ ] My changes look good in both light AND dark mode
-   [ ] The change is not hardcoded to a single network, but has multi-asset in mind
-   [ ] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

<!-- Feel free to add additional comments. -->

samharperpittam reviewed (changes requested):
- Add multiple addresses to profile
- Select "Sign Message"
- Select "Signing Wallet"
- Currently we display a modal to select the signing address, but it should be a dropdown for easier use. See design for dropdown here:

![image](https://github.com/user-attachments/assets/28144016-cd3a-4fb4-ac5c-72d9c40f06c8)

https://www.figma.com/design/OX18z9oCfYiV4cW35bzxsu/ARKVault-Mainsail?node-id=26909-86950&t=3pW3aRfBb0UHUYHM-0

samharperpittam reviewed (changes requested):
After successfully signing a message, the JSON string is empty

![image](https://github.com/user-attachments/assets/54285cdc-284a-422e-bc43-ccebb2733984)

Should look something like this with correctly populated data:
![image](https://github.com/user-attachments/assets/4a395bae-fc85-4f7b-8ee0-9e0915e00f91)

samharperpittam reviewed (changes requested):
Description text should read: `Provide a message below and sign with your mnemonic.`

Currently has `Select an address and sign.`

ItsANameToo commented on src/app/hooks/use-search-parameters-validation.test.ts:
```diff
@@ -218,7 +218,7 @@ describe("useSearchParametersValidation", () => {
 	});
 
-	it("should throw if sign and no message", async () => {
+	it.skip("should throw if sign and no message", async () => {
```
🤔 why are they all skipped

alfonsobries replied:
@ItsANameToo deeplink for signing is not currently working since there's no page to redirect the user to. That task was already too long, and since this needs a refactor of the deeplink handling to open a modal, I figured it was better to add a separate task for that: https://app.clickup.com/t/86dww5u63

samharperpittam reviewed (changes requested):
On the Message Successfully Signed page

- `Signing Wallet` should be `Signing Address`
- `Signature (JSON)` should be 14px
- `Close` and `Copy Signature` buttons should be at the bottom of the page with a border

Design: 
![image](https://github.com/user-attachments/assets/7800a977-6ba1-42bb-b074-cd483d23699b)

alfonsobries commented:
> Description text should read: `Provide a message below and sign with your mnemonic.`
> 
> Currently has `Select an address and sign.`

@samharperpittam the design assumes that an address has already been selected, before that, we don't know if it's going to need a mnemonic, since it could be a Ledger address, for example

shahin-hq commented on src/app/assets/svg/completed-light.svg:
Probably these are already exist in codebase, please check the completed step of sending transfer tx

alfonsobries replied:
@shahin-hq a similar one exists but not the one I need in both dark & light

samharperpittam commented:
> > Description text should read: `Provide a message below and sign with your mnemonic.`
> > Currently has `Select an address and sign.`
> 
> @samharperpittam the design assumes that an address has already been selected, before that, we don't know if it's going to need a mnemonic, since it could be a Ledger address, for example

Yeah fair point. Though, if someone select "Sign Message" when in Single Address view on the Portfolio page, the address should be pre-filled. It's only when in "Multiview" that the address shouldn't be pre-filled.

alfonsobries commented on src/domains/portfolio/pages/Portfolio/Portfolio.tsx:
```diff
@@ -39,9 +42,14 @@ export const Portfolio = () => {
 			</ResetWhenUnmounted>
+
+			<ResetWhenUnmounted>
+				<SignMessageSidePanel open={showSignMessagePanel} onOpenChange={setShowSignMessagePanel} />
```
@shahin-hq just followed the layout of the other components, but you're right—I'll move it to a better place

shahin-hq commented:
@alfonsobries thanks for the fixes, there are two minor issues in UI:

- title height issue in small screen sizes, we either shorten the text to be `Message Signed` or increase the height of header

![image](https://github.com/user-attachments/assets/1f018c4b-b714-4eba-8449-0ded101b855a)

- another issue is `Signature` label in small screen size:

https://www.figma.com/design/OX18z9oCfYiV4cW35bzxsu/ARKVault-Mainsail?node-id=27221-256163&t=HRxB1wxZqZ3OLnQj-0

alfonsobries commented:
@shahin-hq good catches!

Updated the side panel so it handles titles of any length. For that, I had to remove the `fixed` header and footer, and I also refactored how we use it in other side panels like the create and edit wallet panels, which may need some testing to make sure nothing's broken

samharperpittam commented:
https://github.com/user-attachments/assets/1611ce51-0587-4cc6-9c45-36056f735bbe

My final comment on this @alfonsobries  is that if a user tries to manually copy part of the message, it doesn't seem to highlight text correctly which is a bit odd.
